ipatests/test_integration/test_cert.py

```diff
@@ -46,3 +63,43 @@ def test_cacert_file_appear_with_option_F(self):
         assert status == "MONITORING"
 
         self.clients[0].run_command(['ls', '-l', '/etc/pki/tls/test.CA'])
+
+    def test_ipa_getcert_san_aci(self):
+        """Test for DNS and IP SAN extensions + ACIs
+        """
+        hostname = self.clients[0].hostname
+        certfile = '/etc/pki/tls/certs/test2.pem'
+
+        tasks.kinit_admin(self.master)
+        name, zone = hostname.split('.', 1)
+        self.master.run_command(['ipa', 'dnsrecord-show', zone, name])
+        tasks.kdestroy_all(self.master)
+
+        cmd_arg = [
+            'ipa-getcert', 'request', '-v', '-w',
+            '-f', certfile,
+            '-k', '/etc/pki/tls/private/test2.key',
+            '-K', f'test/{hostname}',
+            '-D', hostname,
+            '-A', self.clients[0].ip,
+            # '-D', self.altname,
+        ]
+        result = self.clients[0].run_command(cmd_arg)
+        request_id = re.findall(r'\d+', result.stdout_text)
+
+        # check if certificate is in MONITORING state
+        status = tasks.wait_for_request(self.clients[0], request_id[0], 50)
+        assert status == "MONITORING"
+
+        certdata = self.clients[0].get_file_contents(certfile)
+        cert = x509.load_pem_x509_certificate(
+            certdata, default_backend()
+        )
+        ext = cert.extensions.get_extension_for_oid(
+            ExtensionOID.SUBJECT_ALTERNATIVE_NAME
+        )
+        dnsnames = set(ext.value.get_values_for_type(x509.DNSName))
+        # assert dnsnames == {self.clients[0].hostname, self.altname}
+        assert dnsnames == {self.clients[0].hostname}
+        ipaddrs = set(ext.value.get_values_for_type(x509.IPAddress))
+        assert ipaddrs == {self.clients[0].ip}
```

Comment:
The assert for `ipaddrs` fails because you get `str` in the set on the right side and `IPv4Address` instance on the lest side:
```
>       assert ipaddrs == {self.clients[0].ip}
E       AssertionError: assert {IPv4Address(...168.122.193')} == {'192.168.122.193'}
E         Extra items in the left set:
E         IPv4Address('192.168.122.193')
E         Extra items in the right set:
E         '192.168.122.193'
E         Full diff:
E         - {IPv4Address('192.168.122.193')}
E         + {'192.168.122.193'}
```
I think you either need to convert `self.clients[0].ip` to an IPAddress instance:
```
from ipaddress import ip_address
...
assert ipaddrs == {ip_address(self.clients[0].ip)}
```